mewarpx/mewarpx/assemblies.py

```diff
@@ -106,6 +143,39 @@ def record_scrapedparticles(self):
                     if len(idx_list[-1]) != 0:
                         empty = False
 
+                # sort the particles appropriately if this is an eb
+                if not empty and self.scraper_label == 'eb':
+                    temp_idx_list = []
+                    structs = _libwarpx.get_particle_boundary_buffer_structs(
+                        species.name, self.scraper_label, mwxrun.lev
+                    )
+
+                    if mwxrun.geom_str == 'XZ':
+                        xpos = [struct['x'] for struct in structs]
+                        ypos = [struct['y'] for struct in structs]
+                        zpos = [struct['y'] for struct in structs]
+                    elif mwxrun.geom_str == 'XYZ':
+                        xpos = [struct['x'] for struct in structs]
+                        ypos = [struct['y'] for struct in structs]
+                        zpos = [struct['z'] for struct in structs]
+                    else:
+                        raise RuntimeError(
```

Comment:
```suggestion
                        raise NotImplementedError(
```
This works but I think a NotImplementedError would be more descriptive.